### PR TITLE
Generate javadoc in distribution module

### PR DESCRIPTION
**Pull request by olukas**
Javadoc for Jet+IMDG is currently not generated anywhere in Jet repo. In `3.x` version it was generated in `hazelcast-jet-distribution` but it is not working now since that module is packaged as `pom`.

After this PR javadoc for Jet+IMDG is generated by command `mvn javadoc:aggregate -DskipTests` called from `hazelcast-jet-distribution` module (not from root module).

Moreover version of `maven-javadoc-plugin` is downgraded back to `3.0.1`. Version `3.1.x` has some issue during javadoc generation, version `3.0.1` works correctly. 

Checklist
- [x] Tags Set
- [x] Milestone Set
- [N/A] Any breaking changes are documented
- [N/A] New public APIs have `@Nonnull/@Nullable` annotations
- [N/A] New public APIs have `@since` tags in Javadoc
- [N/A] For code samples, code sample main readme is updated

### CODE REVIEW

**Comment by eminn** [comment]
How about the excluding the `examples` package and shaded (snakeyaml etc.) packages from the javadoc ?

**Comment by olukas** [comment]
@eminn `examples` are not part of javadoc (`hazelcast-jet-distribution` has not dependency on them). I can try to exclude `snakeyaml` and `fasterxml` - they were part of javadoc in previous version (Jet 3.x) but I think we can remove them.

**Comment by olukas** [comment]
@eminn updated, `snakeyaml` and `fasterxml` are not part of generated javadoc any more.

File: pom.xml [hunk 1]
```diff
@@ -80,7 +80,7 @@
         <maven.deploy.plugin.version>2.8.2</maven.deploy.plugin.version>
         <maven.jar.plugin.version>2.4</maven.jar.plugin.version>
         <maven.source.plugin.version>2.2.1</maven.source.plugin.version>
-        <maven.javadoc.plugin.version>3.1.1</maven.javadoc.plugin.version>
```
Comment (cangencer) [inline pom.xml]: why revert to older version?

**Pull request merged**